src/lang/mod.rs

```diff
@@ -295,3 +296,8 @@ fn node_has_ascendants_with_allow_name_attr<'db>(
     }
     false
 }
+
+#[salsa::tracked(returns(ref))]
+fn corelib_context<'db>(db: &'db dyn SemanticGroup) -> CorelibContext<'db> {
+    CorelibContext::new(db)
+}
```

Comment:
nit: would move it above the `node_has_ascendants_with_allow_name_attr`, as it's a query, and the function above is not